### PR TITLE
Query chains deployment state

### PR DESCRIPTION
**Pull request by mdjastrzebski**
Scope:
- [x] Checking etherspot deployment status for all chains
- [x] Modifying account extra for etherspot account to hold `ChainRecord<EtherspotAccount>` instead of single `EtherspotAccount`
- [x] Received actual deployment status for side chains

### CODE REVIEW

File: src/services/etherspot.js [hunk 1]
```diff
@@ -124,8 +118,27 @@ class EtherspotService {
     this.subscription.unsubscribe();
   }
 
-  getAccount(accountAddress: string): Promise<?EtherspotAccount> {
-    return this.sdk.getAccount({ address: accountAddress }).catch((error) => {
+  getSdkForChain(chain: Chain): ?EtherspotSdk {
+    const network = networkNameFromChain(chain);
+    if (!network) {
+      reportErrorLog('EtherspotService getSdkForChain failed', { chain });
```
Comment (poocart) [inline src/services/etherspot.js:124]:
```suggestion
      reportErrorLog('EtherspotService getSdkForChain failed: no network', { chain });
```

File: src/services/etherspot.js [hunk 2]
```diff
@@ -124,8 +118,27 @@ class EtherspotService {
...
+      return undefined;
```
Comment (poocart) [inline src/services/etherspot.js:125]: Maybe just return? And for the one below.

```suggestion
      return;
```

File: src/services/etherspot.js [hunk 3]
```diff
@@ -124,8 +118,27 @@ class EtherspotService {
...
+    }
+
+    const sdk = this.instances[network];
+    if (!sdk) {
+      reportErrorLog('EtherspotService getSdkForChain failed', { chain, network });
```
Comment (poocart) [inline src/services/etherspot.js:130]:
```suggestion
      reportErrorLog('EtherspotService getSdkForChain failed: cannot get SDK instance', { chain, network });
```

File: src/services/etherspot.js [hunk 4]
```diff
@@ -124,8 +118,27 @@ class EtherspotService {
...
+      return undefined;
+    }
+
+    return sdk;
+  }
+
+  getAccount(chain: Chain, accountAddress: string): Promise<?EtherspotAccount> {
+    const sdk = this.getSdkForChain(chain);
+    if (!sdk) return Promise.resolve(null);
```
Comment (poocart) [inline src/services/etherspot.js:139]: It can be just `null` I guess
```suggestion
    if (!sdk) return null;
```

**Review by poocart — APPROVED** [review]
Great work! 🚀  Just a note that app is using Account extra for some other calls such as `getAccountEnsName` which needs to be addressed here as well, DM'd you on this 👌

File: src/services/etherspot.js (hunk 4 above)
Comment (mdjastrzebski) [reply src/services/etherspot.js:139]: Yes, but had to updated return type to `Promise | null`

**Pull request merged**